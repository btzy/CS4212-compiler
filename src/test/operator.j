class Main {

Void main(){
  Int n;Int x;
  n=100;
  x = 10;
  if (n > 99) { println("bigger"); } else { println("not bigger"); }
  if (n > 100) { println("bigger"); } else { println("not bigger"); }
  if (n > 101) { println("bigger"); } else { println("not bigger"); }
  if (n < 99) { println("smaller"); } else { println("not smaller"); }
  if (n < 100) { println("smaller"); } else { println("not smaller"); }
  if (n < 101) { println("smaller"); } else { println("not smaller"); }
  if (n >= 99) { println("bigger or equal"); } else { println("not bigger or equal"); }
  if (n >= 100) { println("bigger or equal"); } else { println("not bigger or equal"); }
  if (n >= 101) { println("bigger or equal"); } else { println("not bigger or equal"); }
  if (n <= 99) { println("smaller or equal"); } else { println("not smaller or equal"); }
  if (n <= 100) { println("smaller or equal"); } else { println("not smaller or equal"); }
  if (n <= 101) { println("smaller or equal"); } else { println("not smaller or equal"); }
  if (n > x) { println("bigger"); } else { println("not bigger"); }
  if (n < x) { println("smaller"); } else { println("not smaller"); }
  if (n >= x) { println("bigger or equal"); } else { println("not bigger or equal"); }
  if (n <= x) { println("smaller or equal"); } else { println("not smaller or equal"); }
  if (n == 99) { println("equal"); } else { println("not equal"); }
  if (n == 100) { println("equal"); } else { println("not equal"); }
  if (n == 101) { println("equal"); } else { println("not equal"); }
  if (n != 99) { println("not equal"); } else { println("equal"); }
  if (n != 100) { println("not equal"); } else { println("equal"); }
  if (n != 101) { println("not equal"); } else { println("equal"); }
  println(n+x);
  println(321);
  println(n+10);
  println(12+n);
  if (99 < n) { println("smaller"); } else { println("not smaller"); }
  if (!(99 < n)) { println("not smaller"); } else { println("smaller"); }
  println("abc"+"defghij");
  println(1+2+3+4+5);
  println(n-x);
  println(x-n);
  println(x*n);
  println(n/x);
  println(x/n);
  println(-x);
  println(n==100 && x==10);
  println(n!=100 && x==10);
  println(n==100 && x!=10);
  println(n!=100 && x!=10);
  println(n==100 || x==10);
  println(n!=100 || x==10);
  println(n==100 || x!=10);
  println(n!=100 || x!=10);
  println(n==100);
  println(!(n==100));
  println(n!=100);
  println(!(n!=100));
  println("test"+null+null+"ab");
}
}
