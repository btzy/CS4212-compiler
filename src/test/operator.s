.text
.global main
main:
stmfd sp!,{a1,fp,lr}
sub sp,sp,#196
mov fp,#100
str fp,[sp,#188]
mov fp,#10
str fp,[sp,#184]
ldr fp,[sp,#188]
mov lr,#99
cmp fp,lr
movgt fp,#1
movle fp,#0
cmp fp,#0
bne .L0
ldr a3,=.LC0
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
b .L1
.L0:
ldr a3,=.LC1
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
.L1:
ldr fp,[sp,#188]
mov lr,#100
cmp fp,lr
movgt fp,#1
movle fp,#0
cmp fp,#0
bne .L2
ldr a3,=.LC0
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
b .L3
.L2:
ldr a3,=.LC1
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
.L3:
ldr fp,[sp,#188]
mov lr,#101
cmp fp,lr
movgt fp,#1
movle fp,#0
cmp fp,#0
bne .L4
ldr a3,=.LC0
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
b .L5
.L4:
ldr a3,=.LC1
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
.L5:
ldr fp,[sp,#188]
mov lr,#99
cmp fp,lr
movlt fp,#1
movge fp,#0
cmp fp,#0
bne .L6
ldr a3,=.LC2
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
b .L7
.L6:
ldr a3,=.LC3
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
.L7:
ldr fp,[sp,#188]
mov lr,#100
cmp fp,lr
movlt fp,#1
movge fp,#0
cmp fp,#0
bne .L8
ldr a3,=.LC2
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
b .L9
.L8:
ldr a3,=.LC3
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
.L9:
ldr fp,[sp,#188]
mov lr,#101
cmp fp,lr
movlt fp,#1
movge fp,#0
cmp fp,#0
bne .L10
ldr a3,=.LC2
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
b .L11
.L10:
ldr a3,=.LC3
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
.L11:
ldr fp,[sp,#188]
mov lr,#99
cmp fp,lr
movge fp,#1
movlt fp,#0
cmp fp,#0
bne .L12
ldr a3,=.LC4
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
b .L13
.L12:
ldr a3,=.LC5
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
.L13:
ldr fp,[sp,#188]
mov lr,#100
cmp fp,lr
movge fp,#1
movlt fp,#0
cmp fp,#0
bne .L14
ldr a3,=.LC4
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
b .L15
.L14:
ldr a3,=.LC5
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
.L15:
ldr fp,[sp,#188]
mov lr,#101
cmp fp,lr
movge fp,#1
movlt fp,#0
cmp fp,#0
bne .L16
ldr a3,=.LC4
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
b .L17
.L16:
ldr a3,=.LC5
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
.L17:
ldr fp,[sp,#188]
mov lr,#99
cmp fp,lr
movle fp,#1
movgt fp,#0
cmp fp,#0
bne .L18
ldr a3,=.LC6
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
b .L19
.L18:
ldr a3,=.LC7
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
.L19:
ldr fp,[sp,#188]
mov lr,#100
cmp fp,lr
movle fp,#1
movgt fp,#0
cmp fp,#0
bne .L20
ldr a3,=.LC6
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
b .L21
.L20:
ldr a3,=.LC7
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
.L21:
ldr fp,[sp,#188]
mov lr,#101
cmp fp,lr
movle fp,#1
movgt fp,#0
cmp fp,#0
bne .L22
ldr a3,=.LC6
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
b .L23
.L22:
ldr a3,=.LC7
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
.L23:
ldr fp,[sp,#188]
ldr lr,[sp,#184]
cmp fp,lr
movgt fp,#1
movle fp,#0
cmp fp,#0
bne .L24
ldr a3,=.LC0
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
b .L25
.L24:
ldr a3,=.LC1
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
.L25:
ldr fp,[sp,#188]
ldr lr,[sp,#184]
cmp fp,lr
movlt fp,#1
movge fp,#0
cmp fp,#0
bne .L26
ldr a3,=.LC2
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
b .L27
.L26:
ldr a3,=.LC3
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
.L27:
ldr fp,[sp,#188]
ldr lr,[sp,#184]
cmp fp,lr
movge fp,#1
movlt fp,#0
cmp fp,#0
bne .L28
ldr a3,=.LC4
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
b .L29
.L28:
ldr a3,=.LC5
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
.L29:
ldr fp,[sp,#188]
ldr lr,[sp,#184]
cmp fp,lr
movle fp,#1
movgt fp,#0
cmp fp,#0
bne .L30
ldr a3,=.LC6
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
b .L31
.L30:
ldr a3,=.LC7
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
.L31:
ldr fp,[sp,#188]
mov lr,#99
cmp fp,lr
moveq fp,#1
movne fp,#0
cmp fp,#0
bne .L32
ldr a3,=.LC8
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
b .L33
.L32:
ldr a3,=.LC9
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
.L33:
ldr fp,[sp,#188]
mov lr,#100
cmp fp,lr
moveq fp,#1
movne fp,#0
cmp fp,#0
bne .L34
ldr a3,=.LC8
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
b .L35
.L34:
ldr a3,=.LC9
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
.L35:
ldr fp,[sp,#188]
mov lr,#101
cmp fp,lr
moveq fp,#1
movne fp,#0
cmp fp,#0
bne .L36
ldr a3,=.LC8
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
b .L37
.L36:
ldr a3,=.LC9
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
.L37:
ldr fp,[sp,#188]
mov lr,#99
cmp fp,lr
movne fp,#1
moveq fp,#0
cmp fp,#0
bne .L38
ldr a3,=.LC9
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
b .L39
.L38:
ldr a3,=.LC8
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
.L39:
ldr fp,[sp,#188]
mov lr,#100
cmp fp,lr
movne fp,#1
moveq fp,#0
cmp fp,#0
bne .L40
ldr a3,=.LC9
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
b .L41
.L40:
ldr a3,=.LC8
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
.L41:
ldr fp,[sp,#188]
mov lr,#101
cmp fp,lr
movne fp,#1
moveq fp,#0
cmp fp,#0
bne .L42
ldr a3,=.LC9
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
b .L43
.L42:
ldr a3,=.LC8
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
.L43:
ldr fp,[sp,#188]
ldr lr,[sp,#184]
add fp,fp,lr
str fp,[sp,#180]
ldr a2,[sp,#180]
ldr a1,=.LZ1
bl printf(PLT)
mov a2,#321
ldr a1,=.LZ1
bl printf(PLT)
ldr fp,[sp,#188]
mov lr,#10
add fp,fp,lr
str fp,[sp,#176]
ldr a2,[sp,#176]
ldr a1,=.LZ1
bl printf(PLT)
mov fp,#12
ldr lr,[sp,#188]
add fp,fp,lr
str fp,[sp,#172]
ldr a2,[sp,#172]
ldr a1,=.LZ1
bl printf(PLT)
mov fp,#99
ldr lr,[sp,#188]
cmp fp,lr
movlt fp,#1
movge fp,#0
cmp fp,#0
bne .L44
ldr a3,=.LC2
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
b .L45
.L44:
ldr a3,=.LC3
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
.L45:
mov fp,#99
ldr lr,[sp,#188]
cmp fp,lr
movlt fp,#1
movge fp,#0
strb fp,[sp,#168]
ldrb fp,[sp,#168]
eor fp,fp,#1
strb fp,[sp,#164]
ldrb fp,[sp,#164]
cmp fp,#0
bne .L46
ldr a3,=.LC3
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
b .L47
.L46:
ldr a3,=.LC2
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
.L47:
ldr fp,=.LC10
ldr fp,[fp,#0]
ldr lr,=.LC11
ldr lr,[lr,#0]
add fp,fp,lr
add a1,fp,#4
bl malloc(PLT)
str fp,[a1,#0]
mov fp,a1
add a1,a1,#4
ldr lr,=.LC10
add a2,lr,#4
ldr a3,[lr,#0]
cmp a3,#0
beq .L49
.L48:
ldrb a4,[a2],#1
strb a4,[a1],#1
subs a3,a3,#1
bne .L48
.L49:
ldr lr,=.LC11
add a2,lr,#4
ldr a3,[lr,#0]
cmp a3,#0
beq .L51
.L50:
ldrb a4,[a2],#1
strb a4,[a1],#1
subs a3,a3,#1
bne .L50
.L51:
str fp,[sp,#160]
ldr a3,[sp,#160]
ldr a2,[a3],#4
ldr a1,=.LZ0
bl printf(PLT)
mov fp,#1
mov lr,#2
add fp,fp,lr
str fp,[sp,#156]
ldr fp,[sp,#156]
mov lr,#3
add fp,fp,lr
str fp,[sp,#152]
ldr fp,[sp,#152]
mov lr,#4
add fp,fp,lr
str fp,[sp,#148]
ldr fp,[sp,#148]
mov lr,#5
add fp,fp,lr
str fp,[sp,#144]
ldr a2,[sp,#144]
ldr a1,=.LZ1
bl printf(PLT)
ldr fp,[sp,#188]
ldr lr,[sp,#184]
sub fp,fp,lr
str fp,[sp,#140]
ldr a2,[sp,#140]
ldr a1,=.LZ1
bl printf(PLT)
ldr fp,[sp,#184]
ldr lr,[sp,#188]
sub fp,fp,lr
str fp,[sp,#136]
ldr a2,[sp,#136]
ldr a1,=.LZ1
bl printf(PLT)
ldr fp,[sp,#184]
ldr lr,[sp,#188]
mul fp,fp,lr
str fp,[sp,#132]
ldr a2,[sp,#132]
ldr a1,=.LZ1
bl printf(PLT)
ldr a1,[sp,#188]
mov a4,a1,ASR#31
add a3,a1,a1,ASR#31
eor a1,a3,a1,ASR#31
ldr a2,[sp,#184]
eor a4,a4,a2,ASR#31
add a3,a2,a2,ASR#31
eor a2,a3,a2,ASR#31
mov a3,#1
.L52:
cmp a1,a2,LSL#1
movcs a2,a2,LSL#1
movcs a3,a3,LSL#1
bcs .L52
mov fp,a4
.L53:
cmp a1,a2
subcs a1,a1,a2
addcs fp,fp,a3
movs a3,a3,LSR#1
movne a2,a2,LSR#1
bne .L53
eor fp,fp,a4
str fp,[sp,#128]
ldr a2,[sp,#128]
ldr a1,=.LZ1
bl printf(PLT)
ldr a1,[sp,#184]
mov a4,a1,ASR#31
add a3,a1,a1,ASR#31
eor a1,a3,a1,ASR#31
ldr a2,[sp,#188]
eor a4,a4,a2,ASR#31
add a3,a2,a2,ASR#31
eor a2,a3,a2,ASR#31
mov a3,#1
.L54:
cmp a1,a2,LSL#1
movcs a2,a2,LSL#1
movcs a3,a3,LSL#1
bcs .L54
mov fp,a4
.L55:
cmp a1,a2
subcs a1,a1,a2
addcs fp,fp,a3
movs a3,a3,LSR#1
movne a2,a2,LSR#1
bne .L55
eor fp,fp,a4
str fp,[sp,#124]
ldr a2,[sp,#124]
ldr a1,=.LZ1
bl printf(PLT)
ldr fp,[sp,#184]
rsb fp,fp,#0
str fp,[sp,#120]
ldr a2,[sp,#120]
ldr a1,=.LZ1
bl printf(PLT)
ldr fp,[sp,#188]
mov lr,#100
cmp fp,lr
moveq fp,#1
movne fp,#0
strb fp,[sp,#116]
ldr fp,[sp,#184]
mov lr,#10
cmp fp,lr
moveq fp,#1
movne fp,#0
strb fp,[sp,#112]
ldrb fp,[sp,#116]
ldrb lr,[sp,#112]
and fp,fp,lr
strb fp,[sp,#108]
ldrb fp,[sp,#108]
cmp fp,#0
ldrne a1,=.LZ2
ldreq a1,=.LZ3
bl printf(PLT)
ldr fp,[sp,#188]
mov lr,#100
cmp fp,lr
movne fp,#1
moveq fp,#0
strb fp,[sp,#104]
ldr fp,[sp,#184]
mov lr,#10
cmp fp,lr
moveq fp,#1
movne fp,#0
strb fp,[sp,#100]
ldrb fp,[sp,#104]
ldrb lr,[sp,#100]
and fp,fp,lr
strb fp,[sp,#96]
ldrb fp,[sp,#96]
cmp fp,#0
ldrne a1,=.LZ2
ldreq a1,=.LZ3
bl printf(PLT)
ldr fp,[sp,#188]
mov lr,#100
cmp fp,lr
moveq fp,#1
movne fp,#0
strb fp,[sp,#92]
ldr fp,[sp,#184]
mov lr,#10
cmp fp,lr
movne fp,#1
moveq fp,#0
strb fp,[sp,#88]
ldrb fp,[sp,#92]
ldrb lr,[sp,#88]
and fp,fp,lr
strb fp,[sp,#84]
ldrb fp,[sp,#84]
cmp fp,#0
ldrne a1,=.LZ2
ldreq a1,=.LZ3
bl printf(PLT)
ldr fp,[sp,#188]
mov lr,#100
cmp fp,lr
movne fp,#1
moveq fp,#0
strb fp,[sp,#80]
ldr fp,[sp,#184]
mov lr,#10
cmp fp,lr
movne fp,#1
moveq fp,#0
strb fp,[sp,#76]
ldrb fp,[sp,#80]
ldrb lr,[sp,#76]
and fp,fp,lr
strb fp,[sp,#72]
ldrb fp,[sp,#72]
cmp fp,#0
ldrne a1,=.LZ2
ldreq a1,=.LZ3
bl printf(PLT)
ldr fp,[sp,#188]
mov lr,#100
cmp fp,lr
moveq fp,#1
movne fp,#0
strb fp,[sp,#68]
ldr fp,[sp,#184]
mov lr,#10
cmp fp,lr
moveq fp,#1
movne fp,#0
strb fp,[sp,#64]
ldrb fp,[sp,#68]
ldrb lr,[sp,#64]
orr fp,fp,lr
strb fp,[sp,#60]
ldrb fp,[sp,#60]
cmp fp,#0
ldrne a1,=.LZ2
ldreq a1,=.LZ3
bl printf(PLT)
ldr fp,[sp,#188]
mov lr,#100
cmp fp,lr
movne fp,#1
moveq fp,#0
strb fp,[sp,#56]
ldr fp,[sp,#184]
mov lr,#10
cmp fp,lr
moveq fp,#1
movne fp,#0
strb fp,[sp,#52]
ldrb fp,[sp,#56]
ldrb lr,[sp,#52]
orr fp,fp,lr
strb fp,[sp,#48]
ldrb fp,[sp,#48]
cmp fp,#0
ldrne a1,=.LZ2
ldreq a1,=.LZ3
bl printf(PLT)
ldr fp,[sp,#188]
mov lr,#100
cmp fp,lr
moveq fp,#1
movne fp,#0
strb fp,[sp,#44]
ldr fp,[sp,#184]
mov lr,#10
cmp fp,lr
movne fp,#1
moveq fp,#0
strb fp,[sp,#40]
ldrb fp,[sp,#44]
ldrb lr,[sp,#40]
orr fp,fp,lr
strb fp,[sp,#36]
ldrb fp,[sp,#36]
cmp fp,#0
ldrne a1,=.LZ2
ldreq a1,=.LZ3
bl printf(PLT)
ldr fp,[sp,#188]
mov lr,#100
cmp fp,lr
movne fp,#1
moveq fp,#0
strb fp,[sp,#32]
ldr fp,[sp,#184]
mov lr,#10
cmp fp,lr
movne fp,#1
moveq fp,#0
strb fp,[sp,#28]
ldrb fp,[sp,#32]
ldrb lr,[sp,#28]
orr fp,fp,lr
strb fp,[sp,#24]
ldrb fp,[sp,#24]
cmp fp,#0
ldrne a1,=.LZ2
ldreq a1,=.LZ3
bl printf(PLT)
ldr fp,[sp,#188]
mov lr,#100
cmp fp,lr
moveq fp,#1
movne fp,#0
strb fp,[sp,#20]
ldrb fp,[sp,#20]
cmp fp,#0
ldrne a1,=.LZ2
ldreq a1,=.LZ3
bl printf(PLT)
ldr fp,[sp,#188]
mov lr,#100
cmp fp,lr
moveq fp,#1
movne fp,#0
strb fp,[sp,#16]
ldrb fp,[sp,#16]
eor fp,fp,#1
strb fp,[sp,#12]
ldrb fp,[sp,#12]
cmp fp,#0
ldrne a1,=.LZ2
ldreq a1,=.LZ3
bl printf(PLT)
ldr fp,[sp,#188]
mov lr,#100
cmp fp,lr
movne fp,#1
moveq fp,#0
strb fp,[sp,#8]
ldrb fp,[sp,#8]
cmp fp,#0
ldrne a1,=.LZ2
ldreq a1,=.LZ3
bl printf(PLT)
ldr fp,[sp,#188]
mov lr,#100
cmp fp,lr
movne fp,#1
moveq fp,#0
strb fp,[sp,#4]
ldrb fp,[sp,#4]
eor fp,fp,#1
strb fp,[sp,#0]
ldrb fp,[sp,#0]
cmp fp,#0
ldrne a1,=.LZ2
ldreq a1,=.LZ3
bl printf(PLT)
mov a1,#0
add sp,sp,#200
ldmfd sp!,{fp,pc}
add sp,sp,#200
ldmfd sp!,{fp,pc}
.data
.align 2
.LC0:
.ascii "\012\000\000\000not bigger"
.align 2
.LC1:
.ascii "\006\000\000\000bigger"
.align 2
.LC2:
.ascii "\013\000\000\000not smaller"
.align 2
.LC3:
.ascii "\007\000\000\000smaller"
.align 2
.LC4:
.ascii "\023\000\000\000not bigger or equal"
.align 2
.LC5:
.ascii "\017\000\000\000bigger or equal"
.align 2
.LC6:
.ascii "\024\000\000\000not smaller or equal"
.align 2
.LC7:
.ascii "\020\000\000\000smaller or equal"
.align 2
.LC8:
.ascii "\011\000\000\000not equal"
.align 2
.LC9:
.ascii "\005\000\000\000equal"
.align 2
.LC10:
.ascii "\003\000\000\000abc"
.align 2
.LC11:
.ascii "\007\000\000\000defghij"
.LZ0:
.asciz "%.*s\012\000"
.LZ1:
.asciz "%d\012\000"
.LZ2:
.asciz "true\012\000"
.LZ3:
.asciz "false\012\000"
