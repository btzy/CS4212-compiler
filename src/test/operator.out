bigger
not bigger
not bigger
not smaller
not smaller
smaller
bigger or equal
bigger or equal
not bigger or equal
not smaller or equal
smaller or equal
smaller or equal
bigger
not smaller
bigger or equal
not smaller or equal
not equal
equal
not equal
not equal
equal
not equal
110
321
110
112
smaller
smaller
abcdefghij
15
90
-90
1000
10
0
-10
true
false
false
false
true
true
true
false
true
false
false
true
